src/node_errors.h

```diff
@@ -75,29 +76,48 @@ void OnFatalError(const char* location, const char* message);
   V(ERR_TLS_INVALID_PROTOCOL_METHOD, TypeError)                                \
   V(ERR_TLS_PSK_SET_IDENTIY_HINT_FAILED, Error)                                \
   V(ERR_VM_MODULE_CACHED_DATA_REJECTED, Error)                                 \
+  V(ERR_VM_MODULE_LINK_FAILURE, Error)                                         \
   V(ERR_WASI_NOT_STARTED, Error)                                               \
   V(ERR_WORKER_INIT_FAILED, Error)                                             \
-  V(ERR_PROTO_ACCESS, Error)                                                   \
+  V(ERR_PROTO_ACCESS, Error)
 
-#define V(code, type)                                                         \
-  inline v8::Local<v8::Value> code(v8::Isolate* isolate,                      \
-                                   const char* message)       {               \
-    v8::Local<v8::String> js_code = OneByteString(isolate, #code);            \
-    v8::Local<v8::String> js_msg = OneByteString(isolate, message);           \
-    v8::Local<v8::Object> e =                                                 \
-        v8::Exception::type(js_msg)->ToObject(                                \
-            isolate->GetCurrentContext()).ToLocalChecked();                   \
-    e->Set(isolate->GetCurrentContext(), OneByteString(isolate, "code"),      \
-           js_code).Check();                                                  \
-    return e;                                                                 \
-  }                                                                           \
-  inline void THROW_ ## code(v8::Isolate* isolate, const char* message) {     \
-    isolate->ThrowException(code(isolate, message));                          \
-  }                                                                           \
-  inline void THROW_ ## code(Environment* env, const char* message) {         \
-    THROW_ ## code(env->isolate(), message);                                  \
+#define V(code, type)                                                          \
+  inline v8::Local<v8::Value> code(                                            \
+      v8::Isolate* isolate, const char* message, va_list args) {               \
+    v8::Local<v8::String> js_code = OneByteString(isolate, #code);             \
+    char buffer[256];                                                          \
```

Comment:
```suggestion
    char buffer[512];                                                          \
```
Does this fix the buffer overflow?

Reply:
vsprintf is inherently unsafe as a generic method (i.e. when message and arguments are not known to be of specific length/type). The solution for problems here would be using vsnprintf, not (just) increasing the buffer size.

If we could use SPrintF and template arguments instead of C-style variadic functions, that would be ideal -- I don't see anything speaking against that here, either

Reply:
wow this is so much better than va args. thank you.